### PR TITLE
Add flutter/assets-for-api-docs to test registry

### PR DESCRIPTION
**Pull request by Piinks**
Fixes https://github.com/flutter/flutter/issues/112116

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: registry/assets_for_api_docs.test [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Tests from https://github.com/flutter/assets-for-api-docs
+
+contact=katelovett@google.com
+fetch=git -c core.longPaths=true clone https://github.com/flutter/assets-for-api-docs.git tests
+fetch=git -c core.longPaths=true -C tests checkout 109051af07cb2927f54512b58d6b25d4f515765b
+update=.
+test=./ci/check --no-activate analyze
```
Comment (Piinks) [inline registry/assets_for_api_docs.test:7]:
```suggestion
test=./ci/check --no-activate analyze

```

Comment (Piinks) [inline registry/assets_for_api_docs.test:7]: @gspencergoog am I holding this right? 🙃 

Also, did you wanted to be listed as a contact?

Comment (gspencergoog) [reply registry/assets_for_api_docs.test:7]: Yes, that should work.  And sure, I'd be happy to be a contact.

Comment (gspencergoog) [reply registry/assets_for_api_docs.test:7]: Although if you don't activate, then you have to make sure that the plugin tools are activated somewhere else.

Comment (Piinks) [reply registry/assets_for_api_docs.test:7]: Oh! whoops, thank you!

Comment (Piinks) [inline registry/assets_for_api_docs.test:7]:
```suggestion
test=./ci/check analyze
```

**Comment by Piinks** [comment]
Circling back here, I am closing this PR for now, the tests registry requires no output from tests, and currently the tests in assets for api docs are very verbose. I'll update the issue to mention we need to add a silent mode or jsut refactor out the noise.

**Pull request closed without merging**